### PR TITLE
Feat stake and delegate

### PR DESCRIPTION
**Pull request by julianmrodri**
* Allow to `stakeAndDelegate` in single transaction

### CODE REVIEW

File: contracts/p1/StRSRVotes.sol [hunk 1]
```diff
@@ -134,6 +134,19 @@ contract StRSRP1Votes is StRSRP1, IStRSRVotes {
...
     }
 
+    /// Stakes an RSR `amount` on the corresponding RToken and allows to delegate
+    /// votes from the sender to `delegatee` or self
+    function stakeAndDelegate(uint256 rsrAmount, address delegatee) external {
+        stake(rsrAmount);
+        address msgSender = _msgSender();
+
+        // Delegate if a specific delegatee is provided or if no delegate yet
+        if (delegatee != address(0) || delegates(msgSender) == address(0)) {
```
Comment (tbrent) [inline contracts/p1/StRSRVotes.sol:144]: No need to re-delegate if the delegate has already been set, so I'm thinking the condition could be `delegatee != address(0) && delegates(msgSender) != delegatee`. Should capture the case that is here and then some.

Comment (tbrent) [reply contracts/p1/StRSRVotes.sol:144]: Hmmm it doesn't capture the delegate-to-self via zero_address, though, does it...

Comment (tbrent) [reply contracts/p1/StRSRVotes.sol:144]: Nvm, LGTM

Comment (julianmrodri) [reply contracts/p1/StRSRVotes.sol]: Implemented the change discussed offline.

**Comment by tbrent** [comment]
LGTM! Can merge once tests finish.

**Pull request merged**